plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go

```diff
@@ -224,7 +224,8 @@ func buildControllerRoles() ([]rbac.ClusterRole, []rbac.ClusterRoleBinding) {
 			// glusterfs
 			rbac.NewRule("get", "list", "watch").Groups(storageGroup).Resources("storageclasses").RuleOrDie(),
 			rbac.NewRule("get", "create", "delete").Groups(legacyGroup).Resources("services", "endpoints").RuleOrDie(),
-			rbac.NewRule("get").Groups(legacyGroup).Resources("secrets").RuleOrDie(),
+			// azure
+			rbac.NewRule("get", "create").Groups(legacyGroup).Resources("secrets").RuleOrDie(),
```

Comment:
This isn't an appropriate permission for the binder. The way it is currently implemented, it risks exposing admin credentials into PVC namespaces

Reply:
The preferred way to accomplish this is to configure the storageclass to specify a namespace for the provisioner to keep its secrets in, and to grant secret permissions in that namespace.

Reply:
@andyzhangx the azure file PV provisioner supports a `secretNamespace` parameter in the storage class. Setting that controls where the provisioner creates the secret. I'd expect the creator of the storage class to:
1. set the `secretNamespace` parameter in the storage class
2. ensure that namespace exists
3. grant secret get/create permissions in that namespace to whatever user is running the azure file provisioner

for example:
```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: azurefile
provisioner: kubernetes.io/azure-file
parameters:
  secretNamespace: azure-file-provisioner
---
apiVersion: v1
kind: Namespace
metadata:
  name: azure-file-provisioner
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: azure-file-provisioner
  namespace: azure-file-provisioner
roleRef:
  kind: ClusterRole
  apiGroup: rbac.authorization.k8s.io
  name: admin
subjects:
- kind: ServiceAccount
  name: persistent-volume-binder
  namespace: kube-system
```